### ISSUE TITLE
Truffle Box that uses a Diamond

### ISSUE THREAD

**Issue by mudgen** [trufflesuite/truffle#3113]
It would be really great if there was a Truffle box that used a Diamond so people would have a nice example of how to use diamonds with Truffle.

"A diamond is a set of contracts that can access the same storage variables and share the same Ethereum address."

The Diamond Standard is here: https://github.com/ethereum/EIPs/issues/2535

**Comment by haltman-at** [comment]
Hm, Truffle boxes are actually under https://github.com/truffle-box , but I guess there's no way to file an issue for an organization as a whole, so maybe this is an OK place for it?  @cds-amal?

**Comment by mudgen** [comment]
Yes, initially looked for a way to open an issue there but I couldn't.

**Comment by cds-amal** [comment]
Hi @mudgen, That's a great idea! Do you have an example box? We would love to add it to our community boxes.

**Comment by mudgen** [comment]
@cds-amal I don't have one.  But the reference implementation for the diamonds is here: https://github.com/ethereum/EIPs/issues/2535

**Comment by fainashalts** [comment]
@mudgen or anyone else who's interested in tackling this, here is a link for how to create a box: https://www.trufflesuite.com/docs/truffle/advanced/creating-a-truffle-box
